conbench/app/compare.py

```diff
@@ -72,7 +72,8 @@ def page(self, comparisons, regressions, improvements, baseline_id, contender_id
                 augment(benchmark)
             outliers, outlier_ids, outlier_names = self.get_outliers(benchmarks)
             outlier_urls = [
-                comparisons_by_id[x]["compare_benchmarks_url"] for x in outlier_ids
+                comparisons_by_id.get(x, {}).get("compare_benchmarks_url", "")
```

Comment:
Yeah... :D those `KeyError`s are obliterated now :fire:. Thanks.

More strictly defined objects will really help make this more sane. 

Just for the record (I believe we all agree): code like this will not be maintainable, and is a rather significant burden for new developers entering a code base. We need to launch an initiative to make it easier again to understand the state space by just reading code. I am writing this now because I hit a situation like this almost daily, and am frequently slowed down spending time on trying to imagine what an object _could be_ during runtime.